### PR TITLE
Create CO.py parser for Colombia's demand

### PR DESCRIPTION
**Pull request by alixunderplatz**
ref #1509 

Creating a parser for the 5-minute real-time demand data for Colombia 🇨🇴 

sample output as of now:
`
[{'zoneKey': 'CO', 'datetime': datetime.datetime(2023, 2, 11, 8, 44, 55, 406000, tzinfo=tzfile('/usr/share/zoneinfo/America/Bogota')), 'consumption': 8218.0, 'storage': {}, 'source': 'xm.com.co'}, {'zoneKey': 'CO', 'datetime': datetime.datetime(2023, 2, 11, 8, 49, 58, 246000, tzinfo=tzfile('/usr/share/zoneinfo/America/Bogota')), 'consumption': 8269.5, 'storage': {}, 'source': 'xm.com.co'}, {'zoneKey': 'CO', 'datetime': datetime.datetime(2023, 2, 11, 8, 55, 1, 87000, tzinfo=tzfile('/usr/share/zoneinfo/America/Bogota')), 'consumption': 8273.9, 'storage': {}, 'source': 'xm.com.co'}, {'zoneKey': 'CO', 'datetime': datetime.datetime(2023, 2, 11, 9, 0, 3, 927000, tzinfo=tzfile('/usr/share/zoneinfo/America/Bogota')), 'consumption': 8254.2, 'storage': {}, 'source': 'xm.com.co'}, {'zoneKey': 'CO', 'datetime': datetime.datetime(2023, 2, 11, 9, 5, 6, 768000, tzinfo=tzfile('/usr/share/zoneinfo/America/Bogota')), 'consumption': 8285.2, 'storage': {}, 'source': 'xm.com.co'}, {'zoneKey': 'CO', 'datetime': datetime.datetime(2023, 2, 11, 9, 10, 9, 608000, tzinfo=tzfile('/usr/share/zoneinfo/America/Bogota')), 'consumption': 8340.9, 'storage': {}, 'source': 'xm.com.co'}]
`

### CODE REVIEW

File: parsers/CO.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    zone_key: str = "CO",
+    session: Session = Session(),
+    target_datetime: Optional[datetime] = None,
+    logger: Logger = getLogger(__name__),
+):
+    if target_datetime:
+        raise NotImplementedError("This parser is not yet able to parse past dates.")
+    with session.get(colombia_demand_URL, verify=False) as response:
+        demand_json = json.loads(response.content)
+        demand_pretty = demand_json['Variables'][0]['Datos']
```
Comment (VIKTORVAV99) [inline parsers/CO.py:27]: I would rename `demand_pretty` to something like `demand_data` just for future readability.

File: parsers/CO.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+        for datapoint in demand_pretty:
+            demand_list.append(
+                {
+                "zoneKey": zone_key,
+                "datetime": arrow.get(datapoint['Fecha'], tzinfo=TZ).datetime,
+                "consumption": round(datapoint['Valor'], 1),
+                "storage": {},
```
Comment (VIKTORVAV99) [inline parsers/CO.py:35]:
```suggestion
```
Storage is not a supported key for consumption.

File: config/zones/CO.yaml [hunk 3]
```diff
@@ -1,2 +1,15 @@
 flag_file_name: co.png
```
Comment (VIKTORVAV99) [inline config/zones/CO.yaml:1]:
```suggestion
```
This is not really related to your pr but this key is no longer used in the frontend so let's remove it 🙂

**Review by VIKTORVAV99 — COMMENTED** [review]
The parser works perfectly but I just have a suggestion about readability and two things we can remove.

We also need to run `poetry run format` for this to get formatted correctly and pass out CI.

File: parsers/CO.py (hunk 1 above)
Comment (alixunderplatz) [reply parsers/CO.py]: fixed :)

File: config/zones/CO.yaml [hunk 4]
```diff
@@ -1,2 +1,15 @@
...
+  gas: 2491
+  hydro: 12549
+  nuclear: 0
+  oil: 1346
+  solar: 297
+  wind: 18
+contributors:
+  - alixunderplatz
+parsers:
+  consumption: CO.fetch_consumption
```
[the remark this answers is not in the document]
Comment (alixunderplatz) [reply config/zones/CO.yaml]: would you be so kind to run this for me? :-) after all thes years working on the map, I still can't understand how and why^^ :-P

[the remark this answers is not in the document]
Comment (VIKTORVAV99) [reply config/zones/CO.yaml]: Sure thing! The why is simple, when we split up the massive json file into all the yaml files so it would be easier to edit and such we also created a script that generated a json file with the keys used in the frontend. But currently it must be generated manually but #5082 fixes that.

But I'll update the files accordingly.

**Review by VIKTORVAV99 — APPROVED** [review]
LGTM! 🎉

Merging this right now!

**Comment by VIKTORVAV99** [comment]
@madsnedergaard looks like the ESLint job is still not working as expected with the required check.
I realize I forgot to talk about this during the meeting but I think we might have to move all of the jobs to a single file as GitHub is treating them as separate workflows when they aren't run in the main CI. Which apparently means the checks stays as expected forever 🫤.
